bridges/modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -319,19 +321,32 @@ impl<T: Config<I>, I: 'static> SendXcm for Pallet<T, I> {
 		dest: &mut Option<MultiLocation>,
 		xcm: &mut Option<Xcm<()>>,
 	) -> SendResult<Self::Ticket> {
-		// we won't have an access to `dest` and `xcm` in the `delvier` method, so precompute
+		// `dest` and `xcm` are required here
+		let dest_ref = dest.as_ref().ok_or(SendError::MissingArgument)?;
+		let xcm_ref = xcm.as_ref().ok_or(SendError::MissingArgument)?;
+
+		// we won't have an access to `dest` and `xcm` in the `deliver` method, so precompute
 		// everything required here
-		let message_size = xcm
-			.as_ref()
-			.map(|xcm| xcm.encoded_size() as _)
-			.ok_or(SendError::MissingArgument)?;
+		let message_size = xcm_ref.encoded_size() as _;
 
 		// bridge doesn't support oversized/overweight messages now. So it is better to drop such
 		// messages here than at the bridge hub. Let's check the message size.
 		if message_size > HARD_MESSAGE_SIZE_LIMIT {
 			return Err(SendError::ExceedsMaxMessageSize)
 		}
 
+		// We need to ensure that the known `dest`'s XCM version can comprehend the current `xcm`
+		// program. This may seem like an additional, unnecessary check, but it is not. A similar
+		// check is probably performed by the `ViaBridgeHubExporter`, which attempts to send a
+		// versioned message to the sibling bridge hub. However, the local bridge hub may have a
+		// higher XCM version than the remote `dest`. Once again, it is better to discard such
+		// messages here than at the bridge hub (e.g., to avoid losing funds).
+		let destination_version = T::DestinationVersion::check_version_for(dest_ref, false)
```

Comment:
Maybe we should remove this `handle_unknown` argument at all. Now it is only `true` inside the `pallet-xcm`, so we could solve it using internal functions.

IIRC `dest` here could point to anything within the bridged chain consensus - e.g. `X3(GlobalConsensus(_), Parachain(_), Account(_))`. And if someone would call it with the `handle_unknown = true`, then we would add an entry to the `VersionDiscoveryQueue`, which in turn would perhaps end up in subscription to remote `Account(_)` version - actually do not know what would happen in that case. So we (in theory) allow flooding the `VersionDiscoveryQueue` with entries of questionable benefit. Of course now it won't happen, because you pass `false`, but it could be changed to `true` later by accidence.

Reply:
@svyatonik @serban300 ok, thank you guys, you're both right, I removed `handle_unknown` and renamed `CheckVersion` to `GetVersion` to keep it simple. So now we only check `dest` version without touching `VersionDiscoveryQueue`.
Anyway, subscription and `VersionDiscoveryQueue` over bridge we will investigate as follow up in Jacob's [issue](https://github.com/paritytech/polkadot-sdk/issues/2417) - added comments there. For now, we rely only on  governance-managed `pallet_xcm::SupportedVersion` configuration.

---

> 2. Here we're validating the initial message without the instructions added by the 

ad. 2) yes, point of the validation here is to validate "initial message" because "initial message" is executed on destination. This is kind of tricky, because `SovereignPaidRemoteExporter` wraps "initial message" only with `ExportMessage(initial_message)` instruction and this is already validate in `XcmpQueue::wrap_version` between source AH -> source BH. At the end, `ExportMessage` is not transferred to the `dest`.  So here we validate "best effort" xcm version between AssetHubs, but on source BH `ExportMessage` is executed and "initial message" is extended about `UniversalOrigin` / `DescendOrigin` by `HaulBlobExporter`. And here in the `pallet-xcm-bridge-hub-router` on source AH we dont know what kind of `HaulBlobExporter` is used on source BH.

So, I would say, that we should rely on the fact that `pallet_xcm::SupportedVersion` is configured by governance for relevant chains correctly - something like Slava said here: https://github.com/paritytech/polkadot-sdk/pull/2481/files#r1415157033. This is how `XcmpQueue` works also.